### PR TITLE
[router] remove store level metrics for non-streaming multiget

### PR DESCRIPTION
**Pull request by xunyin8**
## [router] remove store level metrics for non-streaming multiget

The store level metrics in AggRouterHttpRequestStats are registered lazily. Add a flag in AggRouterHttpRequestStats to disable store level metric emission for non-streaming multiget request type. This is cleaner and the less intrusive than adding checks and handling elsewhere because AggRouterHttpRequestStats and the triggers to corresponding record functions are shared across different request types (SINGLE_GET, MULTI_GET, MULTI_GET_STREAMING, COMPUTE, etc...).

We will keep the total stats for non-streaming multiget to give us visibility in case any legacy clients start to make non-streaming multiget requests.

Fixed a bug where recordNoAvailableReplicaAbortedRetryRequest was calling the wrong method for the store metric recording.

## How was this PR tested?
Existing and new tests

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java [hunk 1]
```diff
@@ -29,14 +32,18 @@ public AggRouterHttpRequestStats(
       ReadOnlyStoreRepository metadataRepository,
       boolean isUnregisterMetricForDeletedStoreEnabled) {
     super(metricsRepository, metadataRepository, isUnregisterMetricForDeletedStoreEnabled);
+    // Disable store level non-streaming multi get stats reporting because it's no longer used in clients. We still
+    // report to the total stats for visibility of potential old clients.
+    isStoreStatsEnabled = !RequestType.MULTI_GET.equals(requestType);
```
Comment (m-nagarajan) [inline services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java:37]: Should we just remove MULTI_GET? Do we have any use cases for this?

File: services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java [hunk 2]
```diff
@@ -46,35 +53,45 @@ public AggRouterHttpRequestStats(
   }
 
   public ScatterGatherStats getScatterGatherStatsForStore(String storeName) {
-    return scatterGatherStatsMap.computeIfAbsent(storeName, k -> new ScatterGatherStats());
+    if (isStoreStatsEnabled) {
+      return scatterGatherStatsMap.computeIfAbsent(storeName, k -> new ScatterGatherStats());
+    } else {
+      return totalScatterGatherStats;
```
Comment (m-nagarajan) [inline services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java:59]: Will this result in dual recording of the total stats?
if we are checking `isStoreStatsEnabled` and not recording the store stats as seen in `recordStoreStats()`, should we clean this to not return anything?

File: services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java (hunk 1 above)
Comment (xunyin8) [reply services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java:37]: Let's remove it when we actually remove the code to support it too.

File: services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java (hunk 2 above)
Comment (xunyin8) [reply services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java]: Good catch, removed and rebased.

File: services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java (hunk 1 above)
Comment (m-nagarajan) [reply services/venice-router/src/main/java/com/linkedin/venice/router/stats/AggRouterHttpRequestStats.java:45]: LGTM

**Review by m-nagarajan — DISMISSED** [review]
Thanks @xunyin8

**Review by m-nagarajan — APPROVED** [review]

**Pull request merged**